test/verify/check-machines-dbus

```diff
@@ -45,5 +45,53 @@ class TestMachinesDBus(machineslib.TestMachines):
         self.provider = "libvirt-dbus"
 
 
+    def testDetachDisk(self):
+        b = self.browser
+        m = self.machine
+
+        # prepare libvirt storage pools
+        m.execute("mkdir /mnt/vm_one; chmod a+rwx /mnt/vm_one")
+        m.execute("virsh pool-create-as myPoolOne --type dir --target /mnt/vm_one")
+        m.execute("virsh vol-create-as myPoolOne mydiskofpoolone_1 --capacity 1G --format qcow2")
+        m.execute("virsh vol-create-as myPoolOne mydiskofpoolone_2 --capacity 1G --format qcow2")
+        wait(lambda: "mydiskofpoolone_1" in m.execute("virsh vol-list myPoolOne"))
+        wait(lambda: "mydiskofpoolone_2" in m.execute("virsh vol-list myPoolOne"))
+
+        self.startVm("subVmTest1")
+
+        m.execute("virsh attach-disk --domain subVmTest1 --source /mnt/vm_one/mydiskofpoolone_1 --target vdb --targetbus virtio")
+        m.execute("virsh attach-disk --domain subVmTest1 --source /mnt/vm_one/mydiskofpoolone_2 --target vdc --targetbus virtio --persistent")
+
+        self.login_and_go("/machines")
+        b.wait_in_text("body", "Virtual Machines")
+        b.wait_present("tbody tr th")
+        b.wait_in_text("tbody tr th", "subVmTest1")
+
+        # Test detaching non permanent disk of a running domain
+        b.click("tbody tr th") # click on the row header
+        b.wait_present("#vm-subVmTest1-state")
+        b.wait_in_text("#vm-subVmTest1-state", "running")
+
+        b.wait_present("#vm-subVmTest1-disks") # wait for the tab
+        b.click("#vm-subVmTest1-disks") # open the "Disks" subtab
+
+        b.wait_present("#vm-subVmTest1-disks-vdb-detach") # button
+        b.click("#vm-subVmTest1-disks-vdb-detach")
+        b.wait_not_present("vm-subVmTest1-disks-vdb-target")
+
+        # Test detaching permanent disk of a stopped domain
+        b.click("#vm-subVmTest1-off-caret")
+        b.wait_visible("#vm-subVmTest1-forceOff")
+        b.click("#vm-subVmTest1-forceOff")
+        b.wait_in_text("#vm-subVmTest1-state", "shut off")
+
+        b.wait_present("#vm-subVmTest1-disks") # wait for the tab
+        b.click("#vm-subVmTest1-disks") # open the "Disks" subtab
+
+        b.wait_present("#vm-subVmTest1-disks-vdc-detach") # button
+        b.click("#vm-subVmTest1-disks-vdc-detach")
+        b.wait_not_present("vm-subVmTest1-disks-vdc-target")
```

Comment:
Is it possible somehow to permanently detach a disk from a running domain? It feels a bit strange to detach a disk, and it will come back as soon as you shut down the VM?

Reply:
Currently I should have the behaviour that the change is permanent always. So if the domain is running we should use both LIVE and CONFIG flags to remove disk from both configuration XMLs.
However there is this case, that a disk might be not present in the inactive XML, so in this case only we should not use the CONFIG flag.

Reply:
I adjusted also the test to reflect this behaviour.